homeassistant/components/obihai/connectivity.py

```diff
@@ -59,7 +59,12 @@ def update(self) -> bool:
             if not self.pyobihai.check_account():
                 return False
 
-        self.serial = self.pyobihai.get_device_serial()
+        self.device_info = {
+            "serial": self.pyobihai.get_device_serial(),
+            "model": self.pyobihai.get_model_name(),
+            "sw_version": self.pyobihai.get_software_version(),
+            "hw_version": self.pyobihai.get_hardware_version(),
+        }
```

Comment:
```suggestion
        self.device_info = DeviceInfo(
            serial=self.pyobihai.get_device_serial(),
            model=self.pyobihai.get_model_name(),
            sw_version=self.pyobihai.get_software_version(),
            hw_version=self.pyobihai.get_hardware_version(),
        )
```

Reply:
I'm confused by the function of this part though, `ObihaiEntity` hass access to these things, and could just as well set this information itself directly? 

I'm not understanding why this is passed?

Reply:
This reduces the number of queries, lookup once, instead of doing the lookup on each entity.

Reply:
There are a couple of reasons for this, the first is that ObihaiEntity is not awaited and these are blocking calls.  update() is awaited during setup, so it's a great time to get our blocking calls out of the way.  I could make these blocking calls in ObihaiEntity, but that leads me to:  The second issue is that this data isn't different for each individual entity, but the same for the entire connection, so doing the queries here reduces the number of times we ask for device_info.  No point in querying for hardware_version for each SIP, and the button, and etc...  It's always the same, so just do it once, and store the result.

I'll also point out that serial is not a valid key for DeviceInfo, which is the primary reason I passed this all as a dict.  One field containing all of the device information I need for ObihaiEntity setup.  I could pass the DeviceInfo object and a separate string containing serial which is needed for the generation of the unique id in ObihaiEntity if you feel it would be more clean to generate the DeviceInfo here in update, then pass it entirely through to ObihaiEntity instead of using a dict.